### PR TITLE
feat(sql): add `nullif(long,long)` SQL function

### PR DESCRIPTION
**Pull request by javier**
NullIf supports now also comparison between long numbers

### CODE REVIEW

**Comment by questdb-butler** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please update the PR title to match this format:
`type(subType): description`

Where `type` is one of:
`feat`, `fix`, `chore`, `docs`, `style`, `refactor`, `perf`, `test`, `ci`, `chore`, `revert`

And: `subType` is one of:
`build`, `sql`, `log`, `mig`, `core`, `ilp`, `pgwire`, `http`, `conf`, `ui`, `wal`

For Example:

```
perf(sql): improve pattern matching performance for SELECT sub-queries
```
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e01c02061c0c18bc3e45334125e4cd32333c2679
</p>

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 21 / 21 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/conditional/NullIfIFunctionFactory.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/conditional/NullIfLongFunctionFactory.java|12|12|100.00%|

File: core/src/main/java/io/questdb/griffin/engine/functions/conditional/NullIfIFunctionFactory.java [hunk 1]
```diff
@@ -30,16 +30,16 @@
 import io.questdb.griffin.FunctionFactory;
 import io.questdb.griffin.PlanSink;
 import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.BinaryFunction;
 import io.questdb.griffin.engine.functions.IntFunction;
-import io.questdb.griffin.engine.functions.UnaryFunction;
 import io.questdb.std.IntList;
 import io.questdb.std.Numbers;
 import io.questdb.std.ObjList;
```
Comment (bziobrowski) [inline core/src/main/java/io/questdb/griffin/engine/functions/conditional/NullIfIFunctionFactory.java:38]: Could you : 
- rename it to NullIfIntFunctionFactory  
- add test with two non-constant int arguments ?

Comment (javier) [reply core/src/main/java/io/questdb/griffin/engine/functions/conditional/NullIfIFunctionFactory.java:38]: Sure. Just pushed the requested changes. Thanks!

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 22 / 22 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/conditional/NullIfIntFunctionFactory.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/conditional/NullIfLongFunctionFactory.java|12|12|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 22 / 22 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/conditional/NullIfIntFunctionFactory.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/conditional/NullIfLongFunctionFactory.java|12|12|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 22 / 22 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/conditional/NullIfIntFunctionFactory.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/conditional/NullIfLongFunctionFactory.java|12|12|100.00%|

**Review by bziobrowski — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 22 / 22 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/conditional/NullIfIntFunctionFactory.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/conditional/NullIfLongFunctionFactory.java|12|12|100.00%|

**Pull request merged**